Additional release notes for Keycloak 24 (#27339)
keycloak/keycloak: docs/documentation/release_notes/topics/24_0_0.adoc

access token. It is *OFF* by default, which means that most claims are not added.

Also, a client policy executor exists. Use it to specify if a particular client request
should use lightweight access tokens or regular access tokens. Alternative to the executor is to use a flag *Always use lightweight access token* on client advanced
settings, which allows to specify that particular client should always use lightweight access tokens. Executor can be an alternative if you need
more flexibility. For instance when you want to use lightweight access tokens by default, but regular tokens just for the specified *scope* parameter etc.

A previous release added an *Add to token introspection* switch. You use it to add
claims that are not present in the access token into the introspection endpoint response.

ifeval::[{project_community}==true]
Thanks to https://github.com/skabano[Shigeyuki Kabano] for the contribution and Thanks to
https://github.com/tnorimat[Takashi Norimatsu] for a help and review of this feature.
endif::[]

== OAuth 2.1 support

> This release contains optional OAuth 2.1 support. New client policy profiles were introduced in this release, which administrator can use to make sure that his clients and particular client requests
> will be compliant with OAuth 2.1 specification. There is dedicated client profile for confidential clients and dedicated profile for public clients.

Reviewer: ```suggestion
This release contains optional OAuth 2.1 support. New client policy profiles were introduced in this release, which administrators can use to make sure that clients and particular client requests comply with the OAuth 2.1 specification. A dedicated client profile exists for confidential clients and a dedicated profile for public clients.
```

ifeval::[{project_community}==true]
Thanks to https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/skabano[Shigeyuki Kabano] for the contribution.
endif::[]

== Scope parameter supported in the refresh token flow

Starting with this release, the *scope* parameter in the OAuth2/OIDC endpoint for token refresh is supported. Use this parameter to request access tokens with a smaller amount
of scopes than originally granted, which means you cannot increase access token scope. This scope limitation does not affect the scope of the refreshed refresh token. This function works as
described in the OAuth2 specification.
ifeval::[{project_community}==true]
Thanks to https://github.com/cgeorgilakis[Konstantinos Georgilakis] for the contribution.
endif::[]

== Client policy executor for secure redirect URIs

> A new client policy executor `secure-redirect-uris-enforcer` is introduced. Use it to restrict what are the redirect URIs, which can be used by the clients. For instance

Reviewer: ```suggestion
A new client policy executor `secure-redirect-uris-enforcer` is introduced. Use it to restrict which redirect URIs can be used by the clients. For instance, 
```

Thanks to https://github.com/MikeTangoEcho[MT] for the contribution.
endif::[]

== OAuth Grant Type SPI

> Internal update in the {project_name} codebase was done to introduce OAuth Grant Type SPI. This allows additional flexibility when introducing custom grant types

Reviewer: ```suggestion
The {project_name} codebase includes an internal update  to introduce the OAuth Grant Type SPI. This update allows additional flexibility when introducing custom grant types
```

> supported by {project_name} OAuth 2 token endpoint.

Reviewer: ```suggestion
supported by the {project_name} OAuth 2 token endpoint.
```
